### PR TITLE
[LOGS] Run logs-agent as a goroutine

### PR DESCRIPTION
**Pull request by ajacquemot**
### What does this PR do?

Run `logs-agent` as a goroutine to be part of the `agent` process and removed all the code to build and run logs-agent in a separate process.

### Motivation

Reuse agents logic (auto-config, status, ...)

### Additional Notes

Other PRs will be opened soon to add a status for `logs-agent`, safely stop and restart `logs-agent` and tackle the `auto-config` topic.

### CODE REVIEW

File: cmd/logs/main.go [hunk 1]
```diff
@@ -1,66 +0,0 @@
...
-				// remove pidfile if set
-				os.Remove(*pidfilePath)
-			}()
-		}
-		Start()
-
-		if config.LogsAgent.GetBool("log_profiling_enabled") {
-			log.Println("starting logs-agent profiling")
-			go func() {
-				log.Println(http.ListenAndServe("localhost:6060", nil))
```
Comment (ajacquemot) [inline cmd/logs/main.go]: @tmichelet why do we need this service to run ?

**Review by mnshdw — DISMISSED** [review]
Looks good !

File: cmd/logs/main.go (hunk 1 above)
Comment (olivielpeau) [reply cmd/logs/main.go]: (note: this was the expvar and profiling port ; now that the logs-agent runs in the main agent process it's handled by the main agent)

**Review by olivielpeau — DISMISSED** [review]
LGTM! 👍

File: cmd/agent/app/start.go [hunk 2]
```diff
@@ -203,6 +204,20 @@ func StartAgent() error {
 	}
 	log.Debugf("statsd started")
 
+	// start logs-agent
+	if config.Datadog.GetBool("log_enabled") {
+		// logs-agent does provide any Stop method yet
```
Comment (masci) [inline cmd/agent/app/start.go]: s/does/doesn't

File: cmd/agent/app/start.go [hunk 3]
```diff
@@ -203,6 +204,20 @@ func StartAgent() error {
...
+		// data loss may happen when stopping agent
```
Comment (masci) [inline cmd/agent/app/start.go]: s/stopping agent/stopping the agent

**Review by masci — DISMISSED** [review]
only a couple of nits, otherwise this LGTM!

**Review by olivielpeau — APPROVED** [review]

**Pull request merged**